src/app/qgshandlebadlayers.cpp

```diff
@@ -552,6 +555,12 @@ void QgsHandleBadLayers::autoFind()
     const QString longName = dataInfo.fileName();
     QString provider = node.namedItem( QStringLiteral( "provider" ) ).toElement().text();
     const QString fileType = mLayerList->item( i, 2 )->text();
+
+    progressDialog.setValue( i );
+    progressDialog.setLabelText( QObject::tr( "Searching for file: %1. \n [ %2 of %3 ] " ).arg( name.left( 15 ) + "...",
```

Comment:
The text here seems to need some fine tuning.

- I'm not sure you wanted to add the three dots in every case (also if the file name is shorter than 15 chars)
- There will right now always be four dots (three appended to the name and one specified in the label template)
- If you stick to the ellipsis use a real ellipsis (`…` instead of `...`)
- For the ellipsis, use a single char (`'…'`)
- I'd probably drop the period `.` from the sentence

Reply:
I went for the distinct dots due to the issues with the ellipsis converting to weird characters, I could use the fix provided by Nyall and directly reference the character code instead of having the ellipsis. 

I also thought about not having them for layers with shorter name but I felt they did not only indicate shortening but also progress/pending action. I'm open to remove them if you feel it is more appropriate.

Reply:
In principle the ellipsis is good to depict truncated names, so using the character code sounds good.

Reply:
Should be fixed now. Had some issues with the layout test and whitespaces, should be all good this time.